docs/book/src/topics/addons.md

```diff
@@ -147,6 +147,138 @@ Apply kube-flannel.yml
+
+Artifacts required for Azure CNI:
+
+- [azure-cni.yaml](https://raw.githubusercontent.com/Azure/azure-container-networking/v1.5.3/hack/manifests/cni-installer-v1.yaml)
+
+### Limitations
+
+- Azure CNI v1 is only supported for Linux nodes. Refer to: [CAPZ#3650](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3650)
+
+- We can only configure one subnet per control-plane node. Refer to: [CAPZ#3506](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3506)
+
+- We can only configure one Network Interface per worker node. Refer to: [Azure-container-networking#3611](https://github.com/Azure/azure-container-networking/issues/1945)
+
+### Update Cluster Configuration
+
+The following resources need to be updated when using `capi-quickstart.yaml` (the default cluster manifest generated while following the Cluster API quick start).
+
+- `kind: AzureCluster`
+  - update `spec.networkSpecs.subnets` with the name and role of the subnets you want to use in your workload cluster.
+
+  - ```yaml
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: AzureCluster
+    metadata:
+      name: ${CLUSTER_NAME}
+      namespace: default
+    spec:
+      .
+      .
+      networkSpec:
+        subnets:
+        - name: control-plane-subnet # update this as per your nomenclature
+          role: control-plane
+        - name: node-subnet # update this as per your nomenclature
+          role: node
+      .
+      .
+    ```
+
+- `kind: KubeadmControlPlane` of control plane nodes
+  - add `max-pods: "110"` to `spec.kubeadmConfigSpec.initConfiguration.nodeRegistration.kubeletExtraArgs`.
+  - add `max-pods: "110"` to `spec.kubeadmConfigSpec.joinConfiguration.nodeRegistration.kubeletExtraArgs`.
+
+  - ```yaml
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    kind: KubeadmControlPlane
+    metadata:
+      name: ${CLUSTER_NAME}-control-plane
+      namespace: default
+    spec:
+      kubeadmConfigSpec:
+        .
+        .
+        initConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              max-pods: "110"
+              .
+              .
+        joinConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              max-pods: "110"
+              .
+              .
+    ```
+
+- `kind: AzureMachineTemplate` of control-plane
+  - Add `networkInterfaces` to controlplane's `AzureMachineTemplate`
+
+  - ```yaml
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: AzureMachineTemplate
+    metadata:
+      name: ${CLUSTER_NAME}-control-plane
+      namespace: default
+    spec:
+      template:
+        spec:
+          .
+          .
+          networkInterfaces:
+          - privateIPConfigs: 30
```

Comment:
the `privateIPConfigs` should match the `maxPods`, so if we use `30` we should use `30` for maxPods everywhere in lieu of 110

Reply:
I was thinking about this too.
Should this be a hard requirement that the MaxIP configured must match the total number of pods that can be allocated on a node?

Reply:
I understand the scenario where `maxIPsPerNode` > `maxPodsPerNode` is valid. Jobs could create multiple ephemeral pods and the cluster would still have free IPs to allocate.

Reply:
However, the scenario where `maxIPsPerNode` < `maxPodsPerNode` does not really makes sense now that I think about it..

Reply:
Updating

Reply:
that's exactly right, if maxPods is larger than IPs allocated the node could run out of IPs

Reply:
Updated